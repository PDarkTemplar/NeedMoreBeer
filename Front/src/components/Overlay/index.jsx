// @flow
import React, { PureComponent } from 'react';
import cn from 'classnames';

import { Transition, animationConstants } from 'common-components/Animation';

import styles from './index.scss';

type Props = {
    noAnimation: boolean,
    visible: boolean,
    className?: string,
};

const enterAnim = {
    opacity: [0, 1],
    duration: 200,
    easing: animationConstants.easing.linear,
};

const exitAnim = {
    opacity: [1, 0],
    duration: 200,
    easing: animationConstants.easing.linear,
};

class Overlay extends PureComponent<Props> {
    static defaultProps = {
        visible: true,
        noAnimation: false,
    };

    renderOverlay = () => {
        const { visible, className } = this.props;

        const overlayClass = cn(className, styles.overlay);

        return visible ? <div className={overlayClass} /> : null;
    };

    render() {
        const { noAnimation } = this.props;

        if (noAnimation) {
            return this.renderOverlay();
        }

        return (
            <Transition enterAnim={enterAnim} exitAnim={exitAnim}>
                {this.renderOverlay()}
            </Transition>
        );
    }
}
export default Overlay;
