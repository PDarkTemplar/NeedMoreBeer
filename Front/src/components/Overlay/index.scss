@import 'imports.scss';

.overlay {
    z-index: 10000;
    border-radius: inherit;
    position: absolute;
    box-sizing: border-box;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(color(white), 0.8);
}
