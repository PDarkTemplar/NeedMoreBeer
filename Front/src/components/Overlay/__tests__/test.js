// @flow
import { shallow } from 'enzyme';
import React from 'react';

import Overlay from '../index';

test('Should render Overlay', () => {
    const overlay = shallow(<Overlay />);

    expect(overlay).toMatchSnapshot();
});

test('Should render Overlay (white)', () => {
    const overlay = shallow(<Overlay color="white" />);

    expect(overlay).toMatchSnapshot();
});

test('Should not render Overlay', () => {
    const overlay = shallow(<Overlay visible={false} />);

    expect(overlay).toMatchSnapshot();
});

test('Should not render static Overlay', () => {
    const overlay = shallow(<Overlay visible={false} static />);

    expect(overlay).toMatchSnapshot();
});

test('Should render static Overlay', () => {
    const overlay = shallow(<Overlay visible static />);

    expect(overlay).toMatchSnapshot();
});
