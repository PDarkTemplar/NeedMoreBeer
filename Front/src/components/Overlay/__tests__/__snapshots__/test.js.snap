// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render Overlay 1`] = `
<AnimeTransition
  enterAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        0,
        1,
      ],
    }
  }
  exitAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        1,
        0,
      ],
    }
  }
/>
`;

exports[`Should not render static Overlay 1`] = `
<AnimeTransition
  enterAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        0,
        1,
      ],
    }
  }
  exitAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        1,
        0,
      ],
    }
  }
/>
`;

exports[`Should render Overlay (white) 1`] = `
<AnimeTransition
  enterAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        0,
        1,
      ],
    }
  }
  exitAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        1,
        0,
      ],
    }
  }
>
  <div
    className="overlay"
  />
</AnimeTransition>
`;

exports[`Should render Overlay 1`] = `
<AnimeTransition
  enterAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        0,
        1,
      ],
    }
  }
  exitAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        1,
        0,
      ],
    }
  }
>
  <div
    className="overlay"
  />
</AnimeTransition>
`;

exports[`Should render static Overlay 1`] = `
<AnimeTransition
  enterAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        0,
        1,
      ],
    }
  }
  exitAnim={
    Object {
      "duration": 200,
      "easing": "linear",
      "opacity": Array [
        1,
        0,
      ],
    }
  }
>
  <div
    className="overlay"
  />
</AnimeTransition>
`;
